### PR TITLE
boards: arm: Add ubx_evknorab10_nrf5340 support

### PR DESCRIPTION
**Pull request by ghost**
Add support for u-blox EVK-NORA-B100 and EVK-NORA-B106
which use the nRF5340. .
This board is functionally equivalent to the nRF5340dk_nrf5340.

Tested with blinky, button, and Bluetooth peripheral_hr

Signed-off-by: Bob Recny <bob.recny@u-blox.com>

### CODE REVIEW

**Comment by brec-u-blox** [comment]
Hi all, just checking on status. Thanks!

**Comment by brec-u-blox** [comment]
@ioannisg, @nashif, @MaureenHelm, @galak 
Hi again. Just curious when this PR might be reviewed. I have a couple more in the works based on the nRF5340 and wanted to get this one approved first.
Thanks!
Bob

**Comment by brec-u-blox** [comment]
Found a small error in nrf5340_cpunet_reset.c - updated.

**Comment by brec-u-blox** [comment]
Just checking in again. It's been some time since the PR was created.

**Comment by brec-u-blox** [comment]
Hi @nashif, @ioannisg, @MaureenHelm, @carlescufi, @galak,
Just checking status.
Thanks,
Bob

**Comment by brec-u-blox** [comment]
Hi @nashif, @ioannisg, @MaureenHelm, @carlescufi, @galak,
Just checking status.
Thanks,
Bob

**Review by MaureenHelm — COMMENTED** [review]
Sorry for the delay. Will you please rebase to the latest main so we can get a fresh CI run?

**Comment by brec-u-blox** [comment]
@MaureenHelm, Thanks - rebased now.

File: boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# u-blox EVK-NORA-B10 board configuration
+
+# Copyright (c) 2021 u-blox AG
+# SPDX-License-Identifier: Apache-2.0
+
+if  BOARD_UBX_EVKNORAB10_NRF5340_CPUAPP || BOARD_UBX_EVKNORAB10_NRF5340_CPUAPP_NS
+
+config BOARD
+	default "ubx_evknorab10_nrf5340_cpuapp" if BOARD_UBX_EVKNORAB10_NRF5340_CPUAPP || BOARD_UBX_EVKNORAB10_NRF5340_CPUAPP_NS
```
Comment (MaureenHelm) [inline boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig:9]: the `if` clause here is redundant to line 6

File: boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+# By default, if we build for a Non-Secure version of the board,
+# force building with TF-M as the Secure Execution Environment.
+
+config BUILD_WITH_TFM
+	# Temporarily disable building Non-Secure images with
+	# TF-M support by default. EVK-NORA-B10 is capable of
+	# TF-M, but not yet added to TrustedFirmware.org as a
+	# supported platform.
```
Comment (MaureenHelm) [inline boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig:18]: Please remove TFM and non-secure related configurations until it can be used

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig (hunk 1 above)
Comment (brec-u-blox) [reply boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig:9]: I was using the nrf5340dk_nrf5340 as a template - same lines are there.

File: boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig (hunk 2 above)
Comment (brec-u-blox) [reply boards/arm/ubx_evknorab10_nrf5340/Kconfig.defconfig:18]: I'll get this added to TrustedFirmware.org soon. Pulled into another project ATM.

**Comment by brec-u-blox** [comment]
I'm going to close this PR, regroup and resubmit after getting the board submitted to trustedfirmware.org. It may be a few more weeks until my schedule frees up to do so. 

My fork of Zephyr has also not been kept current and I'm seeing some strange documentation build errors every day. Evidently, I have other things to sort out first.

**Pull request closed without merging**